application-templates/starter/src/components/channel-details/channel-details.spec.js

```diff
@@ -0,0 +1,338 @@
+        ].filter(Boolean)
+      ),
+    },
+    ...options,
+  });
+  return { history };
+};
+
+const fetchChannelDetailsQueryHandler = graphql.query(
+  'FetchChannelDetails',
+  (_req, res, ctx) => {
+    return res(
+      ctx.data({
+        channel: ChannelMock.random()
+          .key(key)
+          .name(LocalizedString.presets.empty().en(name))
+          .buildGraphql(),
+      })
+    );
+  }
+);
+
+const fetchChannelDetailsQueryHandlerWithNullData = graphql.query(
+  'FetchChannelDetails',
+  (_req, res, ctx) => {
+    return res(ctx.data({ channel: null }));
+  }
+);
+
+const fetchChannelDetailsQueryHandlerWithError = graphql.query(
+  'FetchChannelDetails',
+  (_req, res, ctx) => {
+    return res(
+      ctx.data({ channel: null }),
+      ctx.errors([
+        {
+          message: "Field '$channelId' has wrong value: Invalid ID.",
+        },
+      ])
+    );
+  }
+);
+
+const updateChannelDetailsHandler = graphql.mutation(
+  'UpdateChannelDetails',
+  (_req, res, ctx) => {
+    return res(
+      ctx.data({
+        updateChannel: ChannelMock.random()
+          .key(key)
+          .name(LocalizedString.presets.empty().en(name))
+          .buildGraphql(),
+      })
+    );
+  }
+);
+
+const updateChannelDetailsHandlerWithDuplicateFieldError = graphql.mutation(
+  'UpdateChannelDetails',
+  (_req, res, ctx) => {
+    return res(
+      ctx.data({ updateChannel: null }),
+      ctx.errors([
+        {
+          extensions: {
+            code: 'DuplicateField',
+            field: 'key',
+          },
+        },
+      ])
+    );
+  }
+);
+
+const updateChannelDetailsHandlerWithARandomError = graphql.mutation(
+  'UpdateChannelDetails',
+  (_req, res, ctx) => {
+    return res(
+      ctx.data({ updateChannel: null }),
+      ctx.errors([
+        {
+          message: 'Some fake error message.',
+          code: 'SomeFakeErrorCode',
+        },
+      ])
+    );
+  }
+);
+
+const useMockServerHandlers = (
+  fetchChannelDetailsQueryHandler,
+  updateChannelDetailsMutationHandler
+) => {
+  mockServer.use(
+    ...[
+      graphql.query('FetchChannels', (_req, res, ctx) => {
+        const totalItems = 2;
+
+        return res(
+          ctx.data({
+            channels: buildGraphqlList(
+              Array.from({ length: totalItems }).map((_, index) =>
+                ChannelMock.random().key(`channel-key-${index}`)
+              ),
+              {
+                name: 'Channel',
+                total: totalItems,
+              }
+            ),
+          })
+        );
+      }),
+      fetchChannelDetailsQueryHandler,
+      updateChannelDetailsMutationHandler,
+    ].filter(Boolean)
+  );
+};
+
+describe('rendering', () => {
+  it('should render channel details', async () => {
+    useMockServerHandlers(fetchChannelDetailsQueryHandler);
+    renderApp();
+
+    const keyInput = await screen.findByRole('textbox', {
+      name: /channel key/i,
+    });
```

Comment:
We usually prefer to use `findByLabelText`, which selects the input via the field label.

```suggestion
    const keyInput = await screen.findByLabelText('Channel key');
```